extensions/ql-vscode/src/view/model-editor/MethodRow.tsx

```diff
@@ -103,9 +103,30 @@ const ModelableMethodRow = forwardRef<HTMLElement | undefined, MethodRowProps>(
       onChange,
     } = props;
 
-    const modeledMethods = viewState.showMultipleModels
-      ? modeledMethodsProp
-      : modeledMethodsProp.slice(0, 1);
+    const modeledMethods: Array<ModeledMethod | undefined> = useMemo(
+      () =>
+        modeledMethodsProp.length === 0
+          ? [undefined]
+          : viewState.showMultipleModels
+          ? modeledMethodsProp
+          : modeledMethodsProp.slice(0, 1),
```

Comment:
Nested ternaries are quite hard to read. Could you extract this somehow, for example by lifting the `modeledMethodsProp.length === 0` check out of this ternary and moving it before the ternary?

Reply:
Yep, I agree it's not the prettiest and clearest. I'll extract the whole thing to a method perhaps.